File: regexlib/2021-5-15/javascript_test_file/regexlib_3434.js
/* 3434
 * \d{1,3}[.]\d{1,3}[.]\d{1,3}[.]\d{1,3}\s.\s.\s\[\d{2}\/\D{3}\/\d{4}:\d{1,2}:\d{1,2}:\d{1,2}\s.\d{4}\]\s\&quot;\S*\s\S*\s\S*\&quot;\s\d{1,3}\s\S*\s\&quot;.*\&quot;\s\&quot;.*\&quot;
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"1.1.1.1 1 1 [11/:::/1111:1:1:1 11111] &quot;  &quot; 1  &quot;&quot; &quot;"*5000+"! _1SLQ_2"
 */
var REGEX = new RegExp("\d{1,3}[.]\d{1,3}[.]\d{1,3}[.]\d{1,3}\s.\s.\s\[\d{2}\/\D{3}\/\d{4}:\d{1,2}:\d{1,2}:\d{1,2}\s.\d{4}\]\s\&quot;\S*\s\S*\s\S*\&quot;\s\d{1,3}\s\S*\s\&quot;.*\&quot;\s\&quot;.*\&quot;")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '1.1.1.1 1 1 [11/:::/1111:1:1:1 11111] &quot;  &quot; 1  &quot;&quot; &quot;'.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_3434.py
# 3434
# \d{1,3}[.]\d{1,3}[.]\d{1,3}[.]\d{1,3}\s.\s.\s\[\d{2}\/\D{3}\/\d{4}:\d{1,2}:\d{1,2}:\d{1,2}\s.\d{4}\]\s\&quot;\S*\s\S*\s\S*\&quot;\s\d{1,3}\s\S*\s\&quot;.*\&quot;\s\&quot;.*\&quot;
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"1.1.1.1 1 1 [11/:::/1111:1:1:1 11111] &quot;  &quot; 1  &quot;"*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """\d{1,3}[.]\d{1,3}[.]\d{1,3}[.]\d{1,3}\s.\s.\s\[\d{2}\/\D{3}\/\d{4}:\d{1,2}:\d{1,2}:\d{1,2}\s.\d{4}\]\s\&quot;\S*\s\S*\s\S*\&quot;\s\d{1,3}\s\S*\s\&quot;.*\&quot;\s\&quot;.*\&quot;"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "1.1.1.1 1 1 [11/:::/1111:1:1:1 11111] &quot;  &quot; 1  &quot;" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")